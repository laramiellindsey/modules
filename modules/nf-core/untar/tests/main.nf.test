nextflow_process {

    name "Test Process UNTAR"
    script "../main.nf"
    process "UNTAR"
    tag "untar"
    tag "modules"
    tag "modules_nfcore"

    test("Should run without failures") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                    [],
                    file(params.test_data['sarscov2']['genome']['kraken2_tar_gz'], checkIfExists: true),
                    file(params.test_data['homo_sapiens']['illumina']['test_flowcell'], checkIfExists: true),
                    file(params.test_data['generic']['tar']['tar_gz'], checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll (
            { assert process.success },
            { assert snapshot(process.out).match() }
            )
        }

    }

}
