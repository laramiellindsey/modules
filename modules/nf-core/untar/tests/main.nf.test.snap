{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        [
                            
                        ],
                        [
                            "hash.k2d:md5,8b8598468f54a7087c203ad0190555d9",
                            "opts.k2d:md5,a033d00cf6759407010b21700938f543",
                            "taxo.k2d:md5,094d5891cdccf2f1468088855c214b2c"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,19efec1f58773173ead5b1253b1ecfa9"
                ],
                "untar": [
                    [
                        [
                            
                        ],
                        [
                            "hash.k2d:md5,8b8598468f54a7087c203ad0190555d9",
                            "opts.k2d:md5,a033d00cf6759407010b21700938f543",
                            "taxo.k2d:md5,094d5891cdccf2f1468088855c214b2c"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,19efec1f58773173ead5b1253b1ecfa9"
                ]
            }
        ],
        "timestamp": "2023-10-17T04:19:11.547392"
    }
}